agr_cacher/src/main/java/org/alliancegenome/cacher/cachers/AlleleCacher.java

```diff
@@ -209,39 +211,43 @@ private void readAllFileMetaData() {
 		} catch (Exception e) {
 			e.printStackTrace();
 		}
-		if (downloadSet.getDownloadFileSet() != null) {
-			downloadSet.getDownloadFileSet()
-					.stream().filter(Objects::nonNull)
-					.forEach(source -> source.getFileList()
-							.forEach(file -> {
-								List<String> chromosomes = speciesChromosomeMap.computeIfAbsent(source.getTaxonId(), k -> new ArrayList<>());
-								chromosomes.add(file.getChromosome());
-							}));
+		
+		if (downloadSet.getDownloadFileSources() != null) {
+			for(DownloadSource source: downloadSet.getDownloadFileSources()) {
+				for(String chromosome: source.getChromosomeList()) {
+					List<String> chromosomes = speciesChromosomeMap.computeIfAbsent(source.getTaxonId(), k -> new ArrayList<>());
+					chromosomes.add(chromosome);
+				}
+			}
 		}
 	}
 
 	public void readHtpFiles(String taxonID, String chromosome) {
 		htpAlleleSequenceMap = new ConcurrentHashMap<>();
 		variantMap = new HashMap<>();
-		if (StringUtils.isEmpty(chromosome))
+		if (StringUtils.isEmpty(chromosome)) {
 			return;
+		}
 		try {
-			ExecutorService executor = Executors.newFixedThreadPool(VariantConfigHelper.getSourceDocumentCreatorThreads());
-			DownloadableFile file = downloadSet.getDownloadFileSet().stream()
-					.filter(source -> source.getTaxonId().equals(taxonID))
-					.findAny().get()
-					.getFileList().stream()
-					.filter(downloadableFile -> downloadableFile.getChromosome().equals(chromosome))
-					.findAny().get();
-
-			HtpVariantCreation creator = new HtpVariantCreation(taxonID, chromosome, file, htpAlleleSequenceMap);
-			executor.submit(creator);
-			//Thread.sleep(1000);
-
-			executor.shutdown();
-			while (!executor.isTerminated()) {
-				Thread.sleep(1000);
-			}
+			
+//			ExecutorService executor = Executors.newFixedThreadPool(VariantConfigHelper.getSourceDocumentCreatorThreads());
+//			
```

Comment:
Why is this commented out? Can that be removed?

Reply:
I believe so. But thought you might know this better as I think you wrote it. I see that the HtpVariantCreation thread doesn't actually do anything as it loops through the file.